### PR TITLE
Encrypt lambda environment variables

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: lambda/locals.tf [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
+  api_update_function_name            = "${var.project}-api-update-${local.environment}"
+  checksum_function_name              = "${var.project}-checksum-${local.environment}"
+  create_db_users_function_name       = "${var.project}-create-db-users-${local.environment}"
+  download_files_function_name        = "${var.project}-download-files-${local.environment}"
+  export_api_authoriser_function_name = "${var.project}-export-api-authoriser-${local.environment}"
+  file_format_function_name           = "${var.project}-file-format-${local.environment}"
+  log_data_function_name              = "${var.project}-log-data-${local.environment}"
+  notifications_function_name         = "${var.project}-notifications-${local.environment}"
+  yara_av_function_name               = "${var.project}-yara-av-${local.environment}"
+  api_update_queue_name               = "tdr-api-update-${local.environment}"
```
Comment (TomJKing) [inline lambda/locals.tf:25]: Should the "tdr-" be hardcoded here?

File: lambda/locals.tf [hunk 2]
```diff
@@ -1,28 +1,37 @@
...
+  api_update_queue                    = "arn:aws:sqs:${var.region}:${data.aws_caller_identity.current.account_id}:${local.api_update_queue_name}"
+  api_update_queue_url                = "https://sqs.${var.region}.amazonaws.com/${data.aws_caller_identity.current.account_id}/${local.api_update_queue_name}"
+  antivirus_queue                     = "arn:aws:sqs:${var.region}:${data.aws_caller_identity.current.account_id}:tdr-antivirus-${local.environment}"
```
Comment (TomJKing) [inline lambda/locals.tf:28]: Should the "tdr-" be hardcoded here?

File: lambda/templates/api_update.json.tpl [hunk 3]
```diff
@@ -23,6 +23,14 @@
...
+    {
+      "Sid": "DecryptEnvVar",
+      "Effect": "Allow",
+      "Action": [
+        "kms:Decrypt"
+      ],
+      "Resource": "${kms_arn}"
     }
   ]
 }
```
Comment (TomJKing) [inline lambda/templates/api_update.json.tpl:36]: Missing line break ;)

File: lambda/templates/download_files_lambda.json.tpl [hunk 4]
```diff
@@ -63,6 +63,14 @@
...
+    {
+      "Sid": "DecryptEnvVar",
+      "Effect": "Allow",
+      "Action": [
+        "kms:Decrypt"
+      ],
+      "Resource": "${kms_arn}"
     }
   ]
 }
```
Comment (TomJKing) [inline lambda/templates/download_files_lambda.json.tpl:76]: Missing line break

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments

File: lambda/locals.tf (hunk 1 above)
Comment (MancunianSam) [reply lambda/locals.tf:25]: No, I've swapped it out for `var.project` I've replaced all the other hard coded tdr in this file as well.

File: lambda/locals.tf [hunk 5]
```diff
@@ -22,16 +22,16 @@ locals {
...
-  checksum_queue                      = "arn:aws:sqs:${var.region}:${data.aws_caller_identity.current.account_id}:tdr-checksum-${local.environment}"
-  checksum_queue_url                  = "https://sqs.${var.region}.amazonaws.com/${data.aws_caller_identity.current.account_id}/tdr-checksum-${local.environment}"
-  download_files_queue                = "arn:aws:sqs:${var.region}:${data.aws_caller_identity.current.account_id}:tdr-download-files-${local.environment}"
-  download_files_queue_url            = "https://sqs.${var.region}.amazonaws.com/${data.aws_caller_identity.current.account_id}/tdr-download-files-${local.environment}"
-  file_format_queue                   = "arn:aws:sqs:${var.region}:${data.aws_caller_identity.current.account_id}:tdr-file-format-${local.environment}"
-  file_format_queue_url               = "https://sqs.${var.region}.amazonaws.com/${data.aws_caller_identity.current.account_id}/tdr-file-format-${local.environment}"
+  antivirus_queue                     = "arn:aws:sqs:${var.region}:${data.aws_caller_identity.current.account_id}:${project}-antivirus-${local.environment}"
```
Comment (TomJKing) [inline lambda/locals.tf:28]: Should this not be `${var.project}-...` instead of `${project}`?

**Review by TomJKing — COMMENTED** [review]
One additional comment

File: lambda/locals.tf (hunk 5 above)
Comment (MancunianSam) [reply lambda/locals.tf]: Ah yeah sorry. I've changed it.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**